.circleci/config.yml

```diff
@@ -60,72 +60,98 @@ jobs:
   backend-functional-tests:
     resource_class: large
     docker:
-      - image: cimg/python:3.10
+
+      - image: cimg/python:3.9
         environment:
           SQLALCHEMY_DATABASE_URI: postgresql://taskingmanager@localhost/test_tm
+          POSTGRES_TEST_DB: test_tm
+          POSTGRES_USER: taskingmanager
+          POSTGRES_ENDPOINT: localhost
+
       - image: cimg/postgres:14.2-postgis
         environment:
           POSTGRES_USER: taskingmanager
           POSTGRES_DB: test_tm
+
     steps:
       - checkout
       - setup_remote_docker
-      - run:
-          name: Install packages
-          command: |
-            sudo apt-get update
-            sudo apt-get install -y libgeos-dev # Required for shapely
+      - run: sudo apt-get update
+      - run: sudo apt-get -y install libgeos-dev # Required for shapely
+      - run: sudo apt-get -y install proj-bin libproj-dev
       - run:
           name: Configure Postgresql Test database
           command: |
-            sudo apt-get install postgresql-client
-            psql -h localhost -U $POSTGRES_USER test_$POSTGRES_DB -c "CREATE EXTENSION postgis;"
+            psql \
+            -d $SQLALCHEMY_DATABASE_URI \
+            -c "CREATE EXTENSION postgis;"
+      - run: pip install --upgrade pip
+      - run: pip install -r requirements.txt
       - run:
           name: Run backend functional tests
           command: |
             mkdir ${CIRCLE_WORKING_DIRECTORY}/tests/backend/results
-            chmod -x tests/backend/test*.py
+            find ./tests/backend -name "test*.py" -exec chmod -x {} \;
             nosetests ./tests/backend --with-xunit \
               --xunit-file ${CIRCLE_WORKING_DIRECTORY}/tests/backend/results/unitresults.xml \
               --with-coverage --cover-erase --cover-package=./backend
             coverage xml -o ${CIRCLE_WORKING_DIRECTORY}/tests/backend/results/coverage.xml
+      - store_test_results:
+          path: tests/backend/results/unitresults.xml
+      - store_artifacts:
+          path: tests/backend/results
 
   database-backup:
     resource_class: large
+    parameters:
+      stack_name:
+        description: "Cloudformation stack name"
+        type: string
     docker:
       - image: cimg/postgres:15.1-postgis
```

Comment:
Why does the `database-backup` job use PostgreSQL version 15.1 while all other jobs use version 14.2?

Reply:
Doesn't matter what image is used in this step. It simply runs an RDS backup using AWS API command. Now that you mention it, it could very well be the base image. I will switch it in a different PR. Other jobs use 14.2 to test backend code functionality which requires PostgreSQL version compatibility.